lib/postgrex/pgpass.ex

```diff
@@ -0,0 +1,79 @@
+defmodule Postgrex.Pgpass do
+  @moduledoc """
+  Parses .pgpass file to obtain database credentials
+  """
+
+  use Bitwise, only_operators: true
+
+  @doc """
+  Obtains the password given a keyword list containing hostname, database, port and optionally the username.
+  """
+  def password(opts) do
+    with [_h, _p, _d, _u, password] <- pgpass_for(opts[:hostname], opts[:database], opts[:port], opts[:username], opts[:passfile]) do
+      password
+    else
+      _ -> nil
+    end
+  end
+
+  defp pgpass_for(hostname, database, port, username, passfile) do
+    with {:ok, path} <- readable_pgpass_file?(passfile),
+         {:ok, contents} <- File.read(path),
+         entries when is_list(entries) <- parse(contents) do
+      entries
+      |> filter_hostname(hostname)
+      |> filter_port(port)
+      |> filter_database(database)
+      |> filter_username(username)
+      |> List.first
+    end
+  end
+
+  defp parse(contents) do
+    contents
+    |> String.split("\n")
+    |> Enum.reject(&(String.match?(&1,~r/^#/)))
+    |> Enum.map(&(String.split(&1, ":")))
+  end
+
+  defp readable_pgpass_file?(nil) do
+    # when the pgpass is not passed via the :passfile option, simply ignore and move on
+    with default_path <- pgpass_path(),
+         {:ok, stat} <- File.stat(default_path),
+         0o0600 <- stat.mode &&& 0o0777
+    do {:ok, default_path}
```

Comment:
Put the body on it's own line and move `do` up on the line above. If you use keyword syntax `, do: ` you can do it like this.